src/textual/widgets/_list_view.py

```diff
@@ -154,15 +154,17 @@ def _is_valid_index(self, index: int | None) -> bool:
             return False
         return 0 <= index < len(self._nodes)
 
-    def watch_index(self, old_index: int, new_index: int) -> None:
+    def watch_index(self, old_index: int | None, new_index: int | None) -> None:
         """Updates the highlighting when the index changes."""
         if self._is_valid_index(old_index):
+            assert old_index is not None
             old_child = self._nodes[old_index]
             assert isinstance(old_child, ListItem)
             old_child.highlighted = False
 
         new_child: Widget | None
         if self._is_valid_index(new_index):
+            assert new_index is not None
```

Comment:
Why don't we make `_is_valid_index` a `TypeGuard` so that we can get rid of the `assert x_index is not None`?

Reply:
Good call; curiously I've never seen it used in Textual code.

Reply:
Me neither. Will you do it?

Reply:
Sorry, just saw this. I did it.